dae/dae/genomic_resources/reference_genome.py

```diff
@@ -268,10 +501,122 @@ def calc_info_hash(self):
         return "placeholder"
 
     def calc_statistics_hash(self) -> bytes:
```

Comment:
I would like to discuss if it is possible to make statistics calculations external to the resources. What I think is that we can create `ReferenceGenomeStatisticsBuilder`, which is initialized with the reference genome and organize statistics calculations.

The problem I would like to solve is that when we add statistics calculations in the resource class, the API becomes huge and irrelevant to the usage of the resource itself.

If we agree to work in this direction this task should be separated from the current PR.